src/main/java/com/google/cloud/spanner/pgadapter/wireoutput/DataRowResponse.java

```diff
@@ -15,64 +15,29 @@
 package com.google.cloud.spanner.pgadapter.wireoutput;
 
 import com.google.api.core.InternalApi;
-import com.google.cloud.spanner.ResultSet;
-import com.google.cloud.spanner.pgadapter.ConnectionHandler.QueryMode;
-import com.google.cloud.spanner.pgadapter.ProxyServer.DataFormat;
-import com.google.cloud.spanner.pgadapter.metadata.OptionsMetadata;
-import com.google.cloud.spanner.pgadapter.statements.IntermediateStatement;
 import com.google.cloud.spanner.pgadapter.utils.Converter;
 import java.io.DataOutputStream;
-import java.util.ArrayList;
-import java.util.List;
 
 /** Sends to the client specific row contents. */
 @InternalApi
 public class DataRowResponse extends WireOutput {
-
   private static final int HEADER_LENGTH = 4;
-  private static final int COLUMN_NUMBER_LENGTH = 2;
-  private static final int COLUMN_SIZE_LENGTH = 4;
 
-  private final ResultSet resultSet;
-  private final List<byte[]> columns;
+  private final Converter converter;
 
-  public DataRowResponse(
-      DataOutputStream output,
-      IntermediateStatement statement,
-      ResultSet resultSet,
-      OptionsMetadata options,
-      QueryMode mode) {
+  public DataRowResponse(DataOutputStream output, Converter converter) {
     super(output, 0);
```

Comment:
Instead of moving the whole logic of converting the resultset to byte array and then copying the data from buffer present in the converter object, can we directly write into the outputStream in this class? Maybe something like below:
Instead of 
`columns.add(column);`
we can do
`this.outputStream.write(column, 0, column.length);`

Reply:
That would have been my preference as well, but it is unfortunately not possible. The reason is that the PG wire-protocol requires us to write the length of the message first. In order to get the length of the message, we first need to convert the Spanner row into the PG wire-protocol representation to determine the number of bytes. Then we can write the length, and then we can write the row data.

So to reduce the amount of memory that we need to reserve for this, we do the following:
1. We create a `ByteArrayOutputStream` that serves as a buffer in the `Converter`.
2. The buffer is used to place the converted row. This will allocate as much memory as is needed to convert a given row.
3. The data in the buffer indicates how large the PG wire-protocol representation of the row is, so we can then write the length of the message, and then write the contents of the buffer to the output stream
4. The buffer is reused for each row, meaning that we only need to allocate more memory if a following row is larger than all preceding rows.